TypeCobol.Analysis/Cfg/ControlFlowGraphBuilder.MultiBranchContext.cs

```diff
@@ -35,28 +31,49 @@ internal class MultiBranchContext
             /// The Index in the SuccessorEdge of the RootBlock if any, -1 otherwise.
             /// </summary>
             internal int RootBlockSuccessorIndex;
-
+            /// <summary>
+            /// Terminals block associated to Multi branch Context if any.
+            /// Terminals blocks are block that have as successor NextFlowBlock or
+            /// a branching to the beginning of a loop instruction block.
+            /// </summary>
+            public IList<BasicBlock<Node, D>> Terminals { get; internal set; }
             /// <summary>
             /// The instruction associated to this context.
             /// </summary>
-            internal Node Instruction;
-
+            public Node Instruction { get; internal set; }
             /// <summary>
             /// List to hold When, WhenOther, WhenSearch and AtEnd nodes
             /// encountered in Evaluate or Search statements.
             /// </summary>
             internal List<Node> ConditionNodes;
+            /// <summary>
+            /// Block of the next flow
+            /// </summary>
+            public BasicBlock<Node, D> NextFlowBlock { get; internal set; }
+            /// <summary>
+            /// Any Sub context if any, null otherwise.
+            /// For instance in COBOL EVALUATE Context has all WHEN and WHENOTHER as sub contexts.
+            /// </summary>
+            public IList<IMultiBranchContext<Node, D>> SubContexts { get; internal set; }
+
+            private BasicBlock<Node, D> RootBlockForEnd;
```

Comment:
Can be removed if we also remove ChangeSuccessor.

Reply:
Yes but a keep the code as it is as a reference, if the assertion failed, it will be interresting to know what was inteneded to be done in such situation.

Reply:
done in 96ffab5